From: Tomas Henzl <thenzl@redhat.com>
Date: Sun, 29 Aug 2010 15:49:29 -0400
Subject: [block] cciss: add usage_count attribute to logical drive
Message-id: <1283097002-3341-31-git-send-email-thenzl@redhat.com>
Patchwork-id: 27886
O-Subject: [RHEL6 PATCH 30/63] cciss: Add usage_count attribute to logical
	drives in /sys
Bugzilla: 568830
RH-Acked-by: Neil Horman <nhorman@redhat.com>

Add usage_count attribute to logical drives in /sys

diff --git a/drivers/block/cciss.c b/drivers/block/cciss.c
index e2c9e32..67dba64 100644
--- a/drivers/block/cciss.c
+++ b/drivers/block/cciss.c
@@ -627,6 +627,25 @@ static ssize_t cciss_show_raid_level(struct device *dev,
 }
 DEVICE_ATTR(raid_level, S_IRUGO | S_IWUSR, cciss_show_raid_level, NULL);
 
+static ssize_t cciss_show_usage_count(struct device *dev,
+				      struct device_attribute *attr, char *buf)
+{
+	drive_info_struct *drv = dev_get_drvdata(dev);
+	struct ctlr_info *h = to_hba(drv->dev->parent);
+	unsigned long flags;
+	int count;
+
+	spin_lock_irqsave(CCISS_LOCK(h->ctlr), flags);
+	if (h->busy_configuring) {
+		spin_unlock_irqrestore(CCISS_LOCK(h->ctlr), flags);
+		return -EBUSY;
+	}
+	count = drv->usage_count;
+	spin_unlock_irqrestore(CCISS_LOCK(h->ctlr), flags);
+	return snprintf(buf, 20, "%d\n", count);
+}
+DEVICE_ATTR(usage_count, S_IRUGO, cciss_show_usage_count, NULL);
+
 struct device_attribute *cciss_dev_attrs[] = {
 	&dev_attr_unique_id,
 	&dev_attr_vendor,
@@ -634,6 +653,7 @@ struct device_attribute *cciss_dev_attrs[] = {
 	&dev_attr_rev,
 	&dev_attr_lunid,
 	&dev_attr_raid_level,
+	&dev_attr_usage_count,
 	NULL,
 };
 
